cibuildwheel/options.py

```diff
@@ -506,17 +505,16 @@ def build_options(self, identifier: str | None) -> BuildOptions:
             test_extras = self.reader.get("test-extras", sep=",")
             build_verbosity_str = self.reader.get("build-verbosity")
 
-            build_frontend: BuildFrontend | Literal["default"]
-            if build_frontend_str == "build":
-                build_frontend = "build"
-            elif build_frontend_str == "pip":
-                build_frontend = "pip"
-            elif build_frontend_str == "default":
-                build_frontend = "default"
+            build_frontend_str = self.reader.get("build-frontend", env_plat=False)
```

Comment:
Looks like this needs a `table` kwarg. When I tried this with a `pyproject.toml`, I got the following crash:

```console
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\extra\work\pywinrt\.venv\Scripts\cibuildwheel.exe\__main__.py", line 7, in <module>  
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\__main__.py", line 144, in main
    build_in_directory(args)
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\__main__.py", line 301, in build_in_directory
    print_preamble(platform=platform, options=options, identifiers=identifiers)
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\__main__.py", line 353, in print_preamble
    options_summary = textwrap.indent(options.summary(identifiers), "  ")
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\options.py", line 648, in summary
    build_options = self.build_options(identifier=None)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\options.py", line 508, in build_options
    build_frontend_str = self.reader.get("build-frontend", env_plat=False)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\extra\work\pywinrt\.venv\Lib\site-packages\cibuildwheel\options.py", line 357, in get
    raise ConfigOptionError(msg)
cibuildwheel.options.ConfigOptionError: "'build-frontend' does not accept a table"
```

Reply:
```
build-frontend = "build; args: --skip-dependency-check --no-isolation"
```
does work though, so if that was the intention, then the docs just need to be corrected.

Reply:
Thanks for testing this out! That reveals a hole in my testing coverage too. I'll add a test and a fix.

Reply:
Thanks for the fix. I changed back to the toml inline table syntax and it is working now.